CMakeLists.txt

```diff
@@ -0,0 +1,85 @@
+if (${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+    message(FATAL_ERROR "Prevented in-tree build. Please create a build directory outside of the source code and call cmake from there")
+endif ()
+
+cmake_minimum_required(VERSION 3.9)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+##! Project
+project(magma)
+
+##! Prerequisites CTEST
+enable_testing()
+
+##! Project options
+option(MAGMA_BUILD_TESTS "Build magma tests" OFF)
+option(MAGMA_BUILD_EXAMPLES "Build magma examples" OFF)
+option(IDE_BUILD "Workaround for header-only libraries, put it to ON if you use CLION" OFF)
+
+##! CMake Path
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+include(library)
+
+##! Project modules
+add_subdirectory(modules)
+
+##! Project tests
+if (MAGMA_BUILD_TESTS)
+    add_subdirectory(tests)
+endif ()
+
+##! Project examples
+if (MAGMA_BUILD_EXAMPLES)
+    add_subdirectory(examples)
+endif ()
+
+##! Main library
+add_library(magma INTERFACE)
+
+if (NOT DEFINED ENV{CLAWS_ROOT_DEV})
+    message(STATUS "using insalled version of claws")
+    find_package(claws CONFIG REQUIRED)
+else ()
+    message(STATUS "using development version of claws")
+    add_subdirectory($ENV{CLAWS_ROOT_DEV} build)
+endif ()
+
+target_link_libraries(magma INTERFACE
+        magma::render
+        claws::utils)
```

Comment:
This should be updated since `claws` now also has other modules. Also, you might want to link against `claws` modules inside the `magma` modules that actually need it rather than here